tests/unittests/test_file_accessor.py

```diff
@@ -0,0 +1,92 @@
+        """
+        for mem_map_size in [1, 10, 1024, 2 * 1024 * 1024, 10 * 1024 * 1024]:
+            mem_map_name = self.get_new_mem_map_name()
+            mem_map = self.file_accessor.create_mem_map(mem_map_name,
+                                                        mem_map_size)
+            self.assertIsNotNone(mem_map)
+            delete_status = self.file_accessor.delete_mem_map(mem_map_name,
+                                                              mem_map)
+            self.assertTrue(delete_status)
+
+    def test_create_mem_map_invalid_inputs(self):
+        """
+        Attempt to create memory maps with invalid inputs (size and name) and
+        verify that an SharedMemoryException is raised.
+        """
+        mem_map_name = self.get_new_mem_map_name()
+        inv_mem_map_size = 0
+        with self.assertRaisesRegex(SharedMemoryException, 'Invalid size'):
+            self.file_accessor.create_mem_map(mem_map_name, inv_mem_map_size)
+        inv_mem_map_name = None
+        mem_map_size = 1024
+        with self.assertRaisesRegex(SharedMemoryException, 'Invalid name'):
+            self.file_accessor.create_mem_map(inv_mem_map_name, mem_map_size)
+        inv_mem_map_name = ''
+        with self.assertRaisesRegex(SharedMemoryException, 'Invalid name'):
+            self.file_accessor.create_mem_map(inv_mem_map_name, mem_map_size)
+
+    def test_open_existing_mem_map(self):
+        """
+        Verify that an existing memory map can be opened.
+        """
+        mem_map_size = 1024
+        mem_map_name = self.get_new_mem_map_name()
+        mem_map = self.file_accessor.create_mem_map(mem_map_name, mem_map_size)
+        o_mem_map = self.file_accessor.open_mem_map(mem_map_name, mem_map_size)
+        self.assertIsNotNone(o_mem_map)
+        o_mem_map.close()
+        delete_status = self.file_accessor.delete_mem_map(mem_map_name, mem_map)
+        self.assertTrue(delete_status)
+
+    def test_open_mem_map_invalid_inputs(self):
+        """
+        Attempt to open a memory map with invalid inputs (size and name) and
+        verify that an SharedMemoryException is raised.
+        """
+        mem_map_name = self.get_new_mem_map_name()
+        inv_mem_map_size = -1
+        with self.assertRaisesRegex(SharedMemoryException, 'Invalid size'):
+            self.file_accessor.open_mem_map(mem_map_name, inv_mem_map_size)
+        inv_mem_map_name = None
+        mem_map_size = 1024
+        with self.assertRaisesRegex(SharedMemoryException, 'Invalid name'):
+            self.file_accessor.open_mem_map(inv_mem_map_name, mem_map_size)
+        inv_mem_map_name = ''
+        with self.assertRaisesRegex(SharedMemoryException, 'Invalid name'):
+            self.file_accessor.open_mem_map(inv_mem_map_name, mem_map_size)
+
+    @unittest.skipIf(os.name == 'nt',
+                     'Windows will create an mmap if one does not exist')
+    def test_open_deleted_mem_map(self):
```

Comment:
```python
――――――――――――――――――――――――――――――― TestFileAccessor.test_open_deleted_mem_map ――――――――――――――――――――――――――――――――
Traceback (most recent call last):
  File "/Users/varadmeru/work/microsoft/azfunctions/pyworker/tests/unittests/test_file_accessor.py", line 87, in test_open_deleted_mem_map
    self.assertIsNotNone(o_mem_map)
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 1328, in assertIsNotNone
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/local/Cellar/python@3.8/3.8.6/Frameworks/Python.framework/Versions/3.8/lib/python3.8/unittest/case.py", line 753, in fail
    raise self.failureException(msg)
AssertionError: unexpectedly None
```

Reply:
Same, this should also work. Let's debug this.